apps/meteor/app/lib/server/methods/getMessages.ts

```diff
@@ -16,8 +16,16 @@ Meteor.methods({
 
 		const msgs = Messages.findVisibleByIds(messages).fetch() as IMessage[];
 		const rids = [...new Set(msgs.map((m) => m.rid))];
+		const prids = [
+			...new Set(
+				rids
+					.map((rid) => Rooms.findOneById(rid))
+					.filter((room) => room.prid)
+					.map((room) => room.prid),
```

Comment:
This works fine, but I think it'd be better to use a `reduce` instead of mapping, filtering and mapping again. wdyt?

Reply:
updated!